plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/state/FollowingViewportStateImpl.kt

```diff
@@ -19,6 +20,8 @@ import java.util.concurrent.*
 
 /**
  * The actual implementation of [FollowingViewportState] that follows user's location.
+ *
+ * Note: [LocationComponentPlugin] should be enabled to use this viewport state.
```

Comment:
@macdrevx I also added this note, as by default, we get the location updates from the location puck.

Reply:
what if it's not enabled?

Reply:
there would be no transition, as it's waiting for the first puck location update

Reply:
could we check it somehow and add warning? I was more asking of that

Reply:
We could consider allowing set location provider to this state, and use that to drive the viewport instead of sync with puck.

Reply:
> could we check it somehow and add warning? I was more asking of that

yes, we could check it, I can add some warning Logs